scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/merge-sort-algorithm.py

```diff
@@ -0,0 +1,45 @@
+def merge_sort(array):
+    if not array:
+        return
+    
+    if len(array) > 1:
+        middle_point = len(array) // 2
+        left_part = array[:middle_point]
+        right_part = array[middle_point:]
+
+        merge_sort(left_part)
+        merge_sort(right_part)
+
+        left_array_index = 0
+        right_array_index = 0
+        sorted_index = 0
+
+        while left_array_index < len(left_part) and right_array_index < len(right_part):
+            if left_part[left_array_index] < right_part[right_array_index]:
+                array[sorted_index] = left_part[left_array_index]
+                left_array_index += 1
+            else:
+                array[sorted_index] = right_part[right_array_index]
+                right_array_index += 1
+            sorted_index += 1
+
+        while left_array_index < len(left_part):
+            array[sorted_index] = left_part[left_array_index]
+            left_array_index += 1
+            sorted_index += 1
+
+        while right_array_index < len(right_part):
+            array[sorted_index] = right_part[right_array_index]
+            right_array_index += 1
+            sorted_index += 1
+
+if __name__ == '__main__':
+    numbers = [4, 10, 6, 14, 2, 1, 8, 5]
+
+    print("Unsorted array: ")
+    print(numbers)
+
+    merge_sort(numbers)
+
+    print("Sorted array: ")
+    print(numbers)
```

Comment:
```suggestion
    print(numbers)

```
Nitpick to add blank line at the end of file.